### PR TITLE
FIN-13060 try up to php 8

### PR DESCRIPTION
**Pull request by sergiizeleniy**
ISSUES RESOLVED
-------
https://finsight.myjetbrains.com/youtrack/issue/FIN-13060
https://finsight.myjetbrains.com/youtrack/issue/FIN-13939

### CODE REVIEW

**Review by denys-filchakov — APPROVED** [review]

File: lib/Serializer/POCOSerializer.php [hunk 1]
```diff
@@ -31,21 +31,18 @@
...
-    private $typeDetector;
-    /** @var DependencyResolver */
-    private $resolver;
-    /** @var bool */
-    private $skipNull;
-    /** @var array */
-    private $extendableClasses;
+    private TypeDetector $typeDetector;
+    private DependencyResolver $resolver;
+    private mixed $skipNull;
```
Comment (bogdankoliesnik) [inline lib/Serializer/POCOSerializer.php:44]: bool

File: lib/Serializer/POCOSerializer.php [hunk 2]
```diff
@@ -31,21 +31,18 @@
...
+    private mixed $extendableClasses;
```
Comment (bogdankoliesnik) [inline lib/Serializer/POCOSerializer.php:45]: array

File: lib/Serializer/PrimitiveTypeSerializer.php [hunk 3]
```diff
@@ -61,20 +60,20 @@ public function serialize($data)
         return $data;
     }
 
-    public function unserializeItem($data, Type $type)
+    public function unserializeItem($data, Type $type): float|array|bool|int|string
```
Comment (bogdankoliesnik) [inline lib/Serializer/PrimitiveTypeSerializer.php:63]: it should be mixed
we have `return $data;` if no condition accepted

File: lib/Serializer/DateTimeSerializer.php [hunk 4]
```diff
@@ -38,11 +38,7 @@ public function matches(Type $type): bool
         return is_a($type->getName(), DateTimeInterface::class, true);
     }
 
-    /**
-     * @param DateTimeInterface $data
-     * @return string
-     */
-    public function serialize($data)
+    public function serialize(mixed $data): string
```
Comment (bogdankoliesnik) [inline lib/Serializer/DateTimeSerializer.php:41]: implementations expects $data to be DateTimeInterface

File: lib/Serializer/CollectionSerializer.php [hunk 5]
```diff
@@ -49,12 +48,7 @@ public function matches(Type $type): bool
         return $type->is(Collection::class);
     }
 
-    /**
-     * @param Collection $data
-     *
-     * @return array
-     */
-    public function serialize($data): array
+    public function serialize(mixed $data): array
```
Comment (bogdankoliesnik) [inline lib/Serializer/CollectionSerializer.php:51]: implementations expects $data to be Collection

File: lib/Extractor/ScalarExtractor.php [hunk 6]
```diff
@@ -27,8 +27,7 @@
 
 class ScalarExtractor extends BasicExtractor
 {
-    /** @var int|float|string|bool */
-    private $value;
+    private string|int|bool|float $value;
 
     public function __construct($value)
```
Comment (bogdankoliesnik) [inline lib/Extractor/ScalarExtractor.php:32]: Can we add the same to constructor?
`string|int|bool|float $value`

Comment (sergiizeleniy) [reply lib/Extractor/ScalarExtractor.php:32]: no( 
the next line checks is_scalar

File: lib/Serializer/CollectionSerializer.php (hunk 5 above)
Comment (sergiizeleniy) [reply lib/Serializer/CollectionSerializer.php:51]: Declaration must be compatible with Serializer->serialize(data: mixed)

File: lib/Serializer/DateTimeSerializer.php (hunk 4 above)
Comment (sergiizeleniy) [reply lib/Serializer/DateTimeSerializer.php:41]: Declaration must be compatible with Serializer->serialize(data: mixed)

File: lib/Serializer/POCOSerializer.php (hunk 2 above)
Comment (sergiizeleniy) [reply lib/Serializer/POCOSerializer.php]: done

File: lib/Serializer/POCOSerializer.php (hunk 1 above)
Comment (sergiizeleniy) [reply lib/Serializer/POCOSerializer.php]: done

File: lib/Serializer/PrimitiveTypeSerializer.php (hunk 3 above)
Comment (sergiizeleniy) [reply lib/Serializer/PrimitiveTypeSerializer.php]: done

**Review by bogdankoliesnik — APPROVED** [review]

**Review by denys-filchakov — APPROVED** [review]

**Review by YevhenKyrychenko — APPROVED** [review]

File: .travis.yml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 language: php
 php:
-  - '7.1'
+  - '8.0.13'
```
[the remark this answers is not in the document]
Comment (sergiizeleniy) [reply .travis.yml]: set 8.1.5

**Review by bogdankoliesnik — APPROVED** [review]

**Review by YevhenKyrychenko — APPROVED** [review]

**Review by bogdankoliesnik — APPROVED** [review]

**Review by rsataev — APPROVED** [review]

File: .travis.yml [hunk 8]
```diff
@@ -1,6 +1,6 @@
 language: php
 php:
-  - '7.1'
+  - '8.1.5'
```
Comment (denys-filchakov) [inline .travis.yml:3]: maybe 8.1.6? like in `.github/workflows/psalm-and-phpunit.yml`

Comment (sergiizeleniy) [reply .travis.yml]: fixed

**Review by denys-filchakov — APPROVED** [review]

**Pull request merged**